File: src/Other/Internals/Search/ControlSearch.cs
using System.Collections.Generic;
using System.Linq;

namespace OpenSilver.Samples.Showcase.Search
{
    public static class ControlSearch
    {
        private static List<SearchableItem> controls = SamplesInfoLoader.GetAllControls();

        public static readonly string[] SearchKeywords = controls
            .SelectMany(x => x.Keywords.Append(x.Name))
            .Select(x => x.ToLower().Replace("_demo", ""))
            .Distinct()
            .OrderBy(x => x)
            .ToArray();

        public static List<SearchableItem> Search(string query)
        {
            query = query.ToLower(); // Case insensitive

            return controls.Where(control =>
                control.Name.ToLower().Contains(query) ||  // Match control name
                control.Keywords.Any(k => k.ToLower().Contains(query)) // Match keywords
            ).ToList();
        }
    }
}

File: src/Other/Internals/Search/ControlSearch.vb
Namespace OpenSilver.Samples.Showcase.Search
    Public Class ControlSearch
        Private Shared ReadOnly controls As List(Of SearchableItem) = SamplesInfoLoader.GetAllControls()

        Public Shared ReadOnly SearchKeywords As String() = controls _
            .SelectMany(Function(x) x.Keywords.Append(x.Name)) _
            .Select(Function(x) x.ToLower().Replace("_demo", "")) _
            .Distinct() _
            .OrderBy(Function(x) x) _
            .ToArray()

        Public Shared Function Search(query As String) As List(Of SearchableItem)
            query = query.ToLower() ' Case insensitive

            Return controls.
                Where(Function(control) control.Name.ToLower().Contains(query) OrElse
                      control.Keywords.Any(Function(k) k.ToLower().Contains(query))
                ).ToList()
        End Function
    End Class
End Namespace
